plugins/signalmonitor/signalmonitor.cpp

```diff
@@ -40,14 +40,32 @@
 #include <QTimer>
 
 using namespace GammaRay;
+typedef ServerProxyModel<QSortFilterProxyModel> ServerFilterProxyModel;
+
+class SignalMonitorProxyModel : public ServerFilterProxyModel
+{
+public:
+    using ServerFilterProxyModel::ServerFilterProxyModel;
+
+    void sort(int column, Qt::SortOrder order = Qt::AscendingOrder) override
+    {
+        if (static_cast<SignalHistoryModel::ColumnId>(column) == SignalHistoryModel::MonitoredColumn) {
+            setSortRole(Qt::CheckStateRole);
```

Comment:
I guess you tested this, but I wonder why this isn't triggering a recursion ;-) the proxy is dynamic, and changing the sort role should resort? but apparently not - good to know